core/src/deck.ts

```diff
@@ -0,0 +1,23 @@
+import { Card } from "./card";
+
+// Deck is a collection of cards, the 0th index is the top of the deck
+// The level should be the same for all cards in the deck
+export class Deck {
+  protected _cards: Card[];
+
+  constructor(cards: Card[]) {
+    this._cards = cards;
+  }
+
+  get level() {
+    return this._cards[0].level;
+  }
+
+  get size() {
+    return this._cards.length;
+  }
+
+  draw() {
+    return this._cards.shift() as Card;
+  }
```

Comment:
I don't know if need to add some property on this function or not~? like `get`

Reply:
Out of curiosity, why not make the top of the deck be the last element, so you can use `pop()` instead of `shift()`? I believe using `shift()` requires re-indexing the entire array?

Reply:
oh, is it? would javascript re-index the whole array?

Reply:
[fixed](https://github.com/qoosuperman/panoply/pull/8/commits/ead49f324c119896c5859f08e8442a5dcb67f0d9)